pkg/apis/acme/v1/types_issuer.go

```diff
@@ -303,6 +311,16 @@ type ACMEChallengeSolverHTTP01IngressObjectMeta struct {
 	Labels map[string]string `json:"labels,omitempty"`
 }
 
+type ACMEChallengeSolverHTTP01Istio struct {
+	// Namespace of the gateway that is used to generate the virtualservice for.
+	// +optional
+	GatewayNamespace string `json:"gatewayNamespace,omitempty"`
+
+	// Name of the gateway that is used to generate the virtualservice for.
+	// +optional
+	GatewayName string `json:"gatewayName,omitempty"`
+}
```

Comment:
We should add support here for modifying the solver pod, i.e. `ACMEChallengeSolverHTTP01IngressPodTemplate`.

The issue is the name of that struct (`Ingress`) :sweat_smile: 

This doesn't need to block this feature but will ship with _less_ features than the base implementation.

/cc @munnerz  @maelvls @meyskens @irbekrm

Reply:
I'll add support for modifying the solver pod and keep the "wrong" name.